### DISCUSSION TITLE
Text and face detection at same time

### DISCUSSION THREAD

**Discussion by khawar-islam** [mindee/doctr#968] [Q&A] [answered]
I would like to detect text and faces in documents at same time, I have implemented simple code but it did not workout. 
Any solution?

**Code**

```
for img in glob.glob(testing_path + '*.*'):
    if img.endswith(".pdf"):
        doc = DocumentFile.from_pdf(img)
    else:
        doc = DocumentFile.from_images(img)

    img_bgr_rgb = cv2.imread(img)
    file_Name = os.path.basename(img)

    image = DocumentFile.from_images(img)

    model = ocr_predictor('db_resnet50', 'sar_resnet31', pretrained=True)

    test_face_detector(testing_path)

    result = model(image)

    export = result.export()

    fig = visualize_page(result.pages[0].export(), image[0], interactive=True)
    plt.savefig(os.path.join(result_path, file_Name))
``

**Comment by frgfm** [comment, marked as answer]
Hi again @khawar-islam :wave: 

Thanks for asking! A few things:
- this is not a runnable snippet, you use a `test_face_detector` object but there is no way for us to know what it is :sweat_smile: 
- the  `ocr_predictor` has no connection whatsoever with face detection. We do have a face detection module but that will have to be run separately in your program if "at same time" meant "within the same model"

Please elaborate a bit on your question as mentioned above so that we can help :pray: 

Cheers :v:

**Reply by khawar-islam** [reply]
Hi @frgfm 

thank you for your reply, I have some documents that contains text and face images and i need to detect both things at the same time. I already achieved text part but face detection part is still undetected. **How i can achieve it?**

Cheers

**Reply by frgfm** [reply]
I think you're looking for https://github.com/mindee/doctr/blob/main/doctr/models/artefacts/face.py

I just realized this feature isn't in the documentation, I'll fix this shortly :+1:

**Reply by frgfm** [reply]
While the doc is not available, you only need to instantiate the object and call it :ok_hand: 
It's a wrapper for OpenCV functions

**Reply by khawar-islam** [reply]
@frgfm It works fine, thank you
